helpers/api.go

```diff
@@ -0,0 +1,108 @@
+package helpers
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/cosmos/relayer/relayer"
+	"github.com/gogo/protobuf/proto"
+)
+
+// SuccessJSONResponse prepares data and writes a HTTP success
+func SuccessJSONResponse(status int, v interface{}, w http.ResponseWriter) {
+	out, err := json.Marshal(v)
+	if err != nil {
+		WriteErrorResponse(http.StatusInternalServerError, err, w)
+		return
+	}
+	WriteSuccessResponse(status, out, w)
+}
+
+// SuccessProtoResponse prepares data and writes a HTTP success
+func SuccessProtoResponse(status int, chain *relayer.Chain, v proto.Message, w http.ResponseWriter) {
+	out, err := chain.Encoding.Marshaler.MarshalJSON(v)
+	if err != nil {
+		WriteErrorResponse(http.StatusInternalServerError, err, w)
+		return
+	}
+	WriteSuccessResponse(status, out, w)
+}
+
+// WriteSuccessResponse writes a HTTP success given a status code and data
+func WriteSuccessResponse(statusCode int, data []byte, w http.ResponseWriter) {
+	w.Header().Set("Content-Type", "application/json")
+
+	w.WriteHeader(statusCode)
+
+	w.Write(data)
+}
+
+type errorResponse struct {
+	Err string `json:"err"`
+}
+
+// TODO: do we need better errors
+// errors for things like:
+// - out of funds
+// - transaction errors
+// Lets utilize the codec to make these error returns
+// useful to users and allow them to take proper action
+
+// WriteErrorResponse writes a HTTP error given a status code and an error message
+func WriteErrorResponse(statusCode int, err error, w http.ResponseWriter) {
+	w.Header().Set("Content-Type", "application/json")
+
+	w.WriteHeader(statusCode)
+
+	_ = json.NewEncoder(w).Encode(errorResponse{
```

Comment:
interesting we don't need a w.Write call?

Reply:
We can use w.Write instead of json.NewEncoder but we need to convert errorResponse struct to bytes. So, I just used json.NewEncoder.